canary/producer-cpp/src/Canary.cpp

```diff
@@ -303,226 +156,152 @@ VOID pushErrorMetrics(CustomData *cusData, double duration)
     }
 
     // Send metrics to CW
     cusData->pCWclient->PutMetricDataAsync(cwRequest, onPutMetricDataResponseReceivedHandler);
 }
 
-VOID pushStreamMetrics(CustomData *cusData)
-{    
+// push metric function to publish metrics to cloudwatch after getting g signal from producer sdk cpp
+VOID pushStreamMetrics(CustomData *cusData, KinesisVideoStreamMetrics streamMetrics)
+{
     Aws::CloudWatch::Model::MetricDatum metricDatum;
     Aws::CloudWatch::Model::PutMetricDataRequest cwRequest;
-    cwRequest.SetNamespace("KinesisVideoSDKCanary");    
+    cwRequest.SetNamespace("KinesisVideoSDKCanary");
 
-    auto streamMetrics = cusData->kinesisVideoStream->getMetrics();
-    
     double frameRate = streamMetrics.getCurrentElementaryFrameRate();
     pushMetric("FrameRate", frameRate, Aws::CloudWatch::Model::StandardUnit::Count_Second, metricDatum, cusData->pDimensionPerStream, cwRequest);
     LOG_DEBUG("Frame Rate: " << frameRate);
 
     double transferRate = 8 * streamMetrics.getCurrentTransferRate() / 1024; // *8 makes it bytes->bits. /1024 bits->kilobits
     pushMetric("TransferRate", transferRate, Aws::CloudWatch::Model::StandardUnit::Kilobits_Second,
-        metricDatum, cusData->pDimensionPerStream, cwRequest);
+               metricDatum, cusData->pDimensionPerStream, cwRequest);
     LOG_DEBUG("Transfer Rate: " << transferRate);
 
     double currentViewDuration = streamMetrics.getCurrentViewDuration().count();
     pushMetric("CurrentViewDuration", currentViewDuration, Aws::CloudWatch::Model::StandardUnit::Milliseconds,
-        metricDatum, cusData->pDimensionPerStream, cwRequest);
+               metricDatum, cusData->pDimensionPerStream, cwRequest);
     LOG_DEBUG("Current View Duration: " << currentViewDuration);
 
     if (cusData->pCanaryConfig->useAggMetrics)
     {
         pushMetric("FrameRate", frameRate, Aws::CloudWatch::Model::StandardUnit::Count_Second,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
         pushMetric("TransferRate", transferRate, Aws::CloudWatch::Model::StandardUnit::Kilobits_Second,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
         pushMetric("CurrentViewDuration", currentViewDuration, Aws::CloudWatch::Model::StandardUnit::Milliseconds,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
     }
 
     // Send metrics to CW
     cusData->pCWclient->PutMetricDataAsync(cwRequest, onPutMetricDataResponseReceivedHandler);
 }
 
-VOID pushStartupLatencyMetric(CustomData *data)
+// put frame function to publish metrics to cloudwatch after getting g signal from producer sdk cpp
+static VOID put_frame_kvs(GstElement *kvsSink, VOID *gMetrics, gpointer data)
```

Comment:
Camel case. This is publishing metrics, so maybe just rename to metricHandler. 
Name functions based on functionality. SInce these are callbacks, adding a handler tag would be helpful when looking at the code :)